### PR TITLE
Ncl 1520; NCL-1506; NCL-1509

### PR DESCRIPTION
**Pull request by vibe13**
NCL-1520 - Add option to BuildConfig create page to add it to a BCS.
NCL-1506 - Build configuration page slow loading.
NCL-1509 - Build Configuration Create page is slow to respond.

This set of changes apply the new directives that make use of the infinite-select (and also manage the list of selected components), to the existing dependencies, environments and the newly added BuildGroupConfigs in the BuildConfig create and edit pages. This improves greatly the load of the page, thanks also to the removal of duplicated REST calls. Looking at the REST calls done after these changes, the overall UX experience should be improved greatly.

To improve: the changes to the BuildConfigs links to the BuildGroupConfigs are done in separate calls, so there are multiple notifications appearing when making changes. I believe that they can be improved later, for now it would be good to have and test these optimizations.

@alexcreasy Please review, thanks!

### CODE REVIEW

**Comment by alexcreasy** [comment]
@vibe13 looks great, +1 to merge!

File: rest/src/main/java/org/jboss/pnc/rest/restmodel/ProductMilestoneRest.java [hunk 1]
```diff
@@ -17,17 +17,15 @@
  */
 package org.jboss.pnc.rest.restmodel;
```
Comment (slaskawi) [inline rest/src/main/java/org/jboss/pnc/rest/restmodel/ProductMilestoneRest.java]: This file adds only a noise to the commit. Could you please unstage it?

File: rest/src/main/java/org/jboss/pnc/rest/restmodel/ProductReleaseRest.java [hunk 2]
```diff
@@ -17,22 +17,18 @@
  */
 package org.jboss.pnc.rest.restmodel;
 
-import io.swagger.annotations.ApiModelProperty;
-import org.jboss.pnc.model.ProductMilestone;
-import org.jboss.pnc.model.ProductRelease;
```
Comment (slaskawi) [inline rest/src/main/java/org/jboss/pnc/rest/restmodel/ProductReleaseRest.java]: The same here

File: rest/src/main/java/org/jboss/pnc/rest/restmodel/ProductVersionRest.java [hunk 3]
```diff
@@ -17,24 +17,23 @@
...
-import org.jboss.pnc.rest.validation.groups.WhenCreatingNew;
-import org.jboss.pnc.rest.validation.groups.WhenUpdating;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
 
 import javax.validation.constraints.NotNull;
 import javax.validation.constraints.Null;
 import javax.xml.bind.annotation.XmlRootElement;
-import java.util.ArrayList;
```
Comment (slaskawi) [inline rest/src/main/java/org/jboss/pnc/rest/restmodel/ProductVersionRest.java]: and here ;)

**Comment by slaskawi** [comment]
Only some small issues, but looks good.

**Comment by vibe13** [comment]
@slaskawi @alexcreasy rebased with the latest from master, will merge if CI passes, ok? Thanks!

**Comment by vibe13** [comment]
retest this please

**Comment by vibe13** [comment]
retest this please

**Pull request merged**